### PR TITLE
[web] Fix shadow rendering using boxshadow due to webkit repaint area bug

### PR DESCRIPTION
**Pull request by ferhatb**
## Description

Webkit fails to repaint correct area when filter: blur is used on a dom node.
This PR switches Dom rendering of blurred rectangles to use box shadow instead for webkit.

## Related Issues

https://github.com/flutter/flutter/issues/71835

## Tests

Added dom_mask_filter test.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.
- [x] The reviewer has submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

### CODE REVIEW

File: lib/web_ui/lib/src/engine/dom_canvas.dart [hunk 1]
```diff
@@ -135,6 +135,17 @@ class DomCanvas extends EngineCanvas with SaveElementStackTracking {
   }
 }
 
+/// Converts a shadow color specified by the framework to the color that should
+/// actually be applied when rendering the element.
+///
+/// Returns a color for box-shadow based on blur filter at sigma.
+ui.Color blurColor(ui.Color color, double sigma) {
+  final double strength = math.max(math.min(
+      math.sqrt(sigma) / (math.pi * 2.0), 1.0), 0.0);
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/dom_canvas.dart:144]: Do we need `math.max`? Presumably sigma is >= 0, and so `math.min(math.sqrt(sigma) / (math.pi * 2.0), 1.0)` is also >= 0.

File: lib/web_ui/test/golden_tests/engine/dom_mask_filter_test.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        RecordingCanvas(const Rect.fromLTRB(0, 0, 500, 500));
+    for (int blurSigma = 1; blurSigma < 10; blurSigma += 2) {
+      final Paint paint = Paint()
+        ..color = Color(0xFF2fdfd2)
+        ..maskFilter = MaskFilter.blur(BlurStyle.normal, blurSigma.toDouble());
+      rc.drawRect(Rect.fromLTWH(15.0, 15.0 + blurSigma * 40, 200, 20), paint);
+    }
+    await canvasScreenshot(rc, 'dom_mask_filter_blur',
+        region: screenRect,
+        maxDiffRatePercent: 0.01);
```
Comment (yjbanov) [inline lib/web_ui/test/golden_tests/engine/dom_mask_filter_test.dart:41]: Why not 0 diff rate?

**Review by yjbanov — APPROVED** [review]
lgtm

File: lib/web_ui/test/golden_tests/engine/dom_mask_filter_test.dart (hunk 2 above)
Comment (ferhatb) [reply lib/web_ui/test/golden_tests/engine/dom_mask_filter_test.dart:41]: blur & gradients typically are slightly off between mac and linux.

File: lib/web_ui/lib/src/engine/dom_canvas.dart (hunk 1 above)
Comment (ferhatb) [reply lib/web_ui/lib/src/engine/dom_canvas.dart:144]: Thx. Done.

**Pull request merged**